File: 4. Case Studies/5. Management System/2. InventoryManagement/Entities/Products/Product.cs
using InventoryManagementSystem.Enums;

namespace InventoryManagementSystem.Entities.Products;

public class Product
{
    public string Sku { get; }
    public string Name { get; }
    public double Price { get; }
    public ProductCategory Category { get; }

    public Product(string sku, string name, double price, ProductCategory category)
    {
        Sku = sku;
        Name = name;
        Price = price;
        Category = category;
    }
}

File: 2. Java/4. Case Studies/src/main/java/_5_ManagementSystems/_3_InventoryManagementSystem/entities/Product.java
package _5_ManagementSystems._3_InventoryManagementSystem.entities;

public class Product {
    private final String sku;
    private final String name;
    private final String category;
    private final int lowStockThreshold;

    public Product(String sku, String name, String category, int lowStockThreshold) {
        this.sku = sku;
        this.name = name;
        this.category = category;
        this.lowStockThreshold = lowStockThreshold;
    }

    public String getSku() { return sku;}
    public String getName() { return name;}
    public String getCategory() { return category;}
    public int getLowStockThreshold() {return lowStockThreshold;}
}
